commands/kubernetes.go

```diff
@@ -395,22 +405,169 @@ func RunKubernetesKubeconfigShow(c *CmdConfig) error {
 	return err
 }
 
+func cachedExecCredentialPath(id string) string {
+	return filepath.Join(kubeconfigCachePath(), id+".json")
+}
+
+// loadCachedExecCredential attempts to load the cached exec credential from disk. Never errors
+// Returns nil if there's no credential, if it failed to load it, or if it's expired.
+func loadCachedExecCredential(id string) *clientauthentication.ExecCredential {
+	path := cachedExecCredentialPath(id)
+	f, err := os.Open(path)
+	if err != nil {
+		return nil
+	}
+
+	defer f.Close()
+
+	var execCredential *clientauthentication.ExecCredential
+	if err := json.NewDecoder(f).Decode(&execCredential); err != nil {
+		if Verbose {
+			fmt.Fprintln(os.Stderr, err)
+		}
+		return nil
+	}
+
+	if execCredential.Status == nil {
+		return nil
+	}
+
+	t := execCredential.Status.ExpirationTimestamp
+	if t.IsZero() || t.Time.Before(time.Now()) {
+		if err := os.Remove(path); err != nil && Verbose {
+			fmt.Fprintln(os.Stderr, err)
+		}
+		return nil
+	}
+
+	return execCredential
+}
```

Comment:
I would return an error in this function and log it where you call it and still continue. It's more maintainable in my opionion

Reply:
like with ` -debug` or `-verbose` flag. I'm ok if you don't log it in this PR, but we should be able to do it later.

Reply:
good call